modules/system/controllers/updates/traits/ManagesPlugins.php

```
@@ -0,0 +1,387 @@
+                throw new ApplicationException(Lang::get('system::lang.server.shit_gone_fucky'));
+            }
+
+            /**
+             * @TODO:
+             * - Process the uploaded file to identify the plugins to install
+             * - (optional) require confirmation to install each detected plugin
+             * - Install the identified plugins
+             * - Ensure that deferred binding records and uploaded files are removed post processing or on failure
+             */
+
+            $manager = UpdateManager::instance();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$manager'. (undefined)

(UnusedLocalVariable)